plugins/CoreHome/angularjs/comparisons/comparisons.component.js

```diff
@@ -0,0 +1,144 @@
+            } catch (e) {
+                // ignore
+            }
+        }
+
+        function setUpTooltips() {
+            $element.tooltip({
+                track: true,
+                content: function() {
+                    var title = $(this).attr('title');
+                    return piwikHelper.escape(title.replace(/\n/g, '<br />'));
+                },
+                show: {delay: 200, duration: 200},
+                hide: false
+            });
+        }
+
+        function comparisonHasSegment(comparison) {
+            return typeof comparison.params.segment !== 'undefined';
+        }
+
+        function getComparisonPeriodType(comparison) {
+            var period = comparison.params.period;
+            if (period === 'range') {
+                return _pk_translate('CoreHome_PeriodRange');
+            }
+            var periodStr = _pk_translate('Intl_Period' + period.substring(0, 1).toUpperCase() + period.substring(1));
+            return periodStr.substring(0, 1).toUpperCase() + periodStr.substring(1);
+        }
+
+        function getComparisonTooltip(segmentComparison, periodComparison) {
+            if (!comparisonTooltips
+                || !Object.keys(comparisonTooltips).length
+            ) {
+                return undefined;
+            }
+
+            return comparisonTooltips[periodComparison.index][segmentComparison.index];
+        }
+
+        function onComparisonsChanged() {
+            comparisonTooltips = null;
+
+            if (!vm.comparisonsService.isComparing()) {
+                return;
+            }
+
+            var periodComparisons = comparisonsService.getPeriodComparisons();
+            var segmentComparisons = comparisonsService.getSegmentComparisons();
+            piwikApi.fetch({
+                method: 'API.getProcessedReport',
+                apiModule: 'VisitsSummary',
+                apiAction: 'get',
+                compare: '1',
+                compareSegments: piwikUrl.getSearchParam('compareSegments'),
+                comparePeriods: piwikUrl.getSearchParam('comparePeriods'),
+                compareDates: piwikUrl.getSearchParam('compareDates'),
+                format_metrics: '1',
+            }).then(function (report) {
+                comparisonTooltips = {};
+                periodComparisons.forEach(function (periodComp) {
+                    comparisonTooltips[periodComp.index] = {};
+
+                    segmentComparisons.forEach(function (segmentComp) {
+                        comparisonTooltips[periodComp.index][segmentComp.index] = generateComparisonTooltip(report, periodComp, segmentComp);
+                    });
+                });
+            });
+        }
+
+        function generateComparisonTooltip(visitsSummary, periodComp, segmentComp) {
+            var firstRowIndex = comparisonsService.getComparisonSeriesIndex(periodComp.index, 0);
+            var firstRow = visitsSummary.reportData.comparisons[firstRowIndex];
```

Comment:
`firstRowIndex` will always be set, `getComparisonSeriesIndex()` just does some math.

`visitsSummary.reportData.comparisons` _should_ always be set but I will check in case it isn't.